runhouse/resources/hardware/cluster.py

```diff
@@ -776,11 +783,15 @@ def get(self, key: str, default: Any = None, remote=False):
                 remote=remote,
                 system=self,
             )
+            return res
         except KeyError as e:
             if default == KeyError:
                 raise e
             return default
-        return res
+        except ConnectionError:
```

Comment:
are these errors always guaranteed to be  of type `ConnectionError`?

Reply:
I think so, yes. Because we are sending an HTTP request to the http server on the cluster servlet, but we can't connect to it because the cluster servlet is killed.